packages/enzyme-adapter-utils/src/Utils.js

```diff
@@ -267,18 +267,35 @@ export function simulateError(
   getNodeType = nodeTypeFromType,
   getDisplayName = displayNameOfNode,
 ) {
-  const { componentDidCatch } = catchingInstance || {};
-  if (!componentDidCatch) {
+  const instance = catchingInstance || {};
+
+  const { componentDidCatch } = instance;
+
+  const { getDerivedStateFromError } = instance.constructor;
```

Comment:
similarly, let's pass in the constructor rather than relying on the `.constructor` property